src/dnl/DNL.h

```diff
@@ -0,0 +1,210 @@
+// SPDX-FileCopyrightText: 2024 The Naja authors <https://github.com/xtofalex/naja/blob/main/AUTHORS>
+//
+// SPDX-License-Identifier: Apache-2.0
+
+#include <fstream>
+#include <iostream>
+#include <set>
+#include <stack>
+#include <vector>
+#include "SNLUniverse.h"
+#include "tbb/scalable_allocator.h"
+
+using namespace naja::SNL;
+
+namespace naja {
+namespace SNL {
+    class SNLBitNet;
+}
+} 
+
+namespace naja {
+namespace DNL {
+
+typedef size_t DNLID;
+#define DNLID_MAX ((DNLID)-1)
+
+class DNL;
+class DNLTerminal;
+
+class DNLInstance {
+ public:
+    DNLInstance();
+    DNLInstance(const SNLInstance* instance, DNLID id, DNLID parent);
+    void display() const;
+    DNLID getID() const;
+    DNLID getParentID() const;
+    const DNLInstance& getParentInstance() const;
+    const SNLInstance* getSNLInstance() const;
+    const SNLDesign* getSNLModel() const;
+    void setTermsIndexes(const std::pair<DNLID, DNLID>& termsIndexes);
+    void setChildrenIndexes(const std::pair<DNLID, DNLID>& childrenIndexes);
+    const DNLInstance& getChildInstance(const SNLInstance* snlInst) const;
+    const DNLTerminal& getTerminal(const SNLInstTerm* snlTerm) const;
+    const DNLTerminal& getTerminalFromBitTerm(const SNLBitTerm* snlTerm) const;
+    const std::pair<DNLID, DNLID>& getTermIndexes() const {
+        return _termsIndexes;
+    }
+    bool isNull() const { return _id == (DNLID)DNLID_MAX; }
+    bool isTop() const { return _parent == (DNLID)DNLID_MAX;; }
+    const std::pair<DNLID, DNLID>& getChildren() const {
+        return _childrenIndexes;
+    }
+    bool isLeaf() const {
+        return _childrenIndexes.first == _childrenIndexes.second;
+    }
+
+ private:
+    std::pair<DNLID, DNLID> _childrenIndexes;
+    const SNLInstance* _instance = nullptr;
+    DNLID _id = DNLID_MAX;
+    DNLID _parent = DNLID_MAX;
+    std::pair<DNLID, DNLID> _termsIndexes;
+};
+
+class DNLTerminal {
+ public:
+    DNLTerminal(DNLID id);
+    DNLTerminal(DNLID DNLInstID, SNLInstTerm* terminal, DNLID id);
+    DNLTerminal(DNLID DNLInstID, SNLBitTerm* terminal, DNLID id);
+    DNLID getID() const;
+    SNLInstTerm* getSnlTerm() const;
+    SNLBitTerm* getSnlBitTerm() const;
+    const DNLInstance& getDNLInstance() const;
+    bool isNull() const { return _id == (DNLID)DNLID_MAX; }
+    void setIsoID(DNLID isoID);
+    DNLID getIsoID() const;
+    bool isTopPort() const { return _terminal == nullptr; }
+
+ private:
+    SNLInstTerm* _terminal = nullptr;
+    SNLBitTerm* _bitTerminal = nullptr;
```

Comment:
use the {} initialization:
SNLBitTerm* bitTerminal_ { nullptr };
for private members